internal/controller/networking/ingress_controller.go

```diff
@@ -52,81 +54,126 @@ func (r *IngressReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	logger := log.FromContext(ctx)
 	logger.Info("Reconciler started")
 
-	ingress := &networkingv1.Ingress{}
-	apiCheck := &checklyv1alpha1.ApiCheck{}
-
+	// ////////////////////////////////
+	// Setup
+	// ///////////////////////////////
+	ingress := networkingv1.Ingress{}
+	// apiCheck := &checklyv1alpha1.ApiCheck{}
 	annotationEnabled := fmt.Sprintf("%s/enabled", r.ControllerDomain)
-
+	checklyFinalizer := fmt.Sprintf("%s/finalizer", r.ControllerDomain)
 	// Check if ingress object is still present
-	err := r.Get(ctx, req.NamespacedName, ingress)
+	err := r.Get(ctx, req.NamespacedName, &ingress)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			logger.Info("Ingress got deleted")
-			return ctrl.Result{}, nil
+			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 		logger.Error(err, "Can't read the Ingress object")
 		return ctrl.Result{}, err
 	}
 	logger.Info("Ingress Object found")
 
-	// Check if annotation is present on the object
-	checklyAnnotation := ingress.Annotations[annotationEnabled] == "true"
-	if !checklyAnnotation {
-		// Annotation may have been removed or updated, we have to determine if we need to delete a previously created ApiCheck resource
-		logger.Info("annotation is not present, checking if ApiCheck was created")
-		err = r.Get(ctx, req.NamespacedName, apiCheck)
-		if err != nil {
-			logger.Info("Apicheck not present")
+	// Gather data for the checkly check
+	logger.Info("Gathering data for the check")
+	apiCheckResources, err := r.gatherApiCheckData(&ingress)
+	if err != nil {
+		logger.Info("unable to gather data for the apiCheck resource", "Ingress Name", ingress.Name, "Ingress namespace", ingress.Namespace)
+		return ctrl.Result{}, err
+	}
+
+	// Do we want to do anything with the ingress?
+	_, checklyAnnotationExists := ingress.Annotations[annotationEnabled]
+	if (!checklyAnnotationExists) || (ingress.Annotations[annotationEnabled] == "false") {
```

Comment:
Seems kinda complex. Given how the exists variable is single use here, you might as well just put them in the `if`'s scope. That way you can give them short generic names and it's not going to pollute entire scope. Also, minus one map access (my favorite).

```suggestion
	if value, exists := ingress.Annotations[annotationEnabled]; !exists || value == "false" {
```

Reply:
Fixed in latest commit.